tests/lib/reflash.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# flash.sh: Install a flashing process to reboot to a new disk image
+#
+# Copyright (C) 2023-2024 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+# This produces an Ubuntu Core image for testing in file `pc.img.gz`.
+# It will initialize the spread user as well as the project directory
+# (which has to be defined in `PROJECT_PATH`).
+# This is expected to be called from spread.
+
+# This is to reflash a spread runner.
+# This script takes one argument: a gzipped image to be flashed onto the disk.
+# After successfully running this script, a reboot or power down will
+# flash the machine with the given image.
+
+set -eux
+
+DEBIAN_FRONTEND=noninteractive apt install -y --no-install-recommends dracut-core busybox-initramfs
+
+[ -d /run/initramfs ] || mkdir -p /run/initramfs
+
+systemd-mount --property=DefaultDependencies=no --options=exec,size=2G none --type=tmpfs /run/initramfs
+
+cp -T "${1}" /run/initramfs/image.gz
+
+for try in /usr/lib/systemd/systemd-shutdown /lib/systemd/systemd-shutdown; do
+    if [ -x "${try}" ]; then
+        systemd_shutdown="${try}"
+    fi
+done
+
+/usr/lib/dracut/dracut-install --ldd -D/run/initramfs -a umount systemctl dd "${systemd_shutdown}"
+/usr/lib/dracut/dracut-install -D/run/initramfs /usr/lib/initramfs-tools/bin/busybox /bin/busybox
```

Comment:
is busybox from initrafms tools a static binary or are all deps satisfied by already?

Reply:
Right, I am missing the `--ldd`. It only needs `libc.so.6`, so that works, but let's add it.